src/System.Collections.Specialized/src/System/Collections/Specialized/ListDictionary.cs

```diff
@@ -493,6 +492,7 @@ public void Reset()
         }
 
         [Serializable]
+        [System.Runtime.CompilerServices.TypeForwardedFrom("System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
```

Comment:
I guess the only real way to know is to debug through the BinaryFormatter code and see if it looks at the TypeForwardedFrom attributes on nested types.